src/input.rs

```diff
@@ -0,0 +1,116 @@
+use crate::model::*;
+use egg::*;
+use std::collections::HashMap;
+
+/// Struct for converting a model specified using our Rust interface to RecExpr
+///
+/// The RecExpr is growed on the fly when member functions are called. Uses a
+/// Hashmap to store the map of scalar nodes to their indices into the RexExpr to
+/// avoid replication.
+#[derive(Default)]
+pub struct GraphConverter {
+    rec_expr: RecExpr<Mdl>,
```

Comment:
Might be good to document these here - also are all of these public? E.g., is there expectation that name_gen will be called outside the struct?

Reply:
These fields are private, they are not going to be called outside.